src/zlib_h.jl

```diff
@@ -1,11 +1,14 @@
 # general zlib constants, definitions
 
-@unix_only    const _zlib = "libz"
-@windows_only const _zlib = "zlib1"
+@static if is_unix()
+    const _zlib = "libz"
+elseif iswindows()
```

Comment:
This currently only exists on Julia v0.4 and needs to be added to Compat.jl.

Cc: @vtjnash